### ISSUE TITLE
Support GZip for FullHttpResponse

### ISSUE THREAD

**Issue by allenxwang** [ReactiveX/RxNetty#35]
We need to add another configuration option in ClientConfig to indicate that GZip support is needed and add appropriate handler in the pipeline.

**Comment by NiteshKant** [comment]
Is this fixed or not required?

**Comment by NiteshKant** [comment]
Yeah we should add this feature. Although, is there any specific reason why this should only apply to fullhttpresponse?

**Comment by allenxwang** [comment]
From Netty's examples, looks like this needs to be after the HttpClientCodec. 

So not specific to fullhttpresponse, but probably does not make sense for ServerSentEvent without chunked encoding.

**Comment by NiteshKant** [comment]
I see what you are saying. The example uses the http compression/decompression codec which in turn uses codecs in this package: https://github.com/netty/netty/blob/4.0/codec/src/main/java/io/netty/handler/codec/compression/package-info.java# I think we should aim at modeling client/server compression using this package. HTTP will be a specific use case as it in turn means adding compression type headers. 
What this means is that compression/decompression can be a generic attribute applied for any client/server.
